KeenClient/KeenClient.m

```diff
@@ -412,6 +446,16 @@ - (void)locationManager:(CLLocationManager *)manager
     }
 }
 
+-(void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
+    if ([KeenClient isLocationAuthorized:status]) {
+        [self startMonitoringLocation];
+    }
+}
+-(void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *)error
+{
+  
```

Comment:
@Liron-K just adding a `KCLog` statement is fine. At least that can help with debugging